File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/20-Set-Map/LeetCode/LeetCode-0001-1.java
class Solution {
    /// O(n^2) algorithm with O(1) extra space
    public int[] twoSum(int[] nums, int target) {
        for (int i=0; i<nums.length-1; i++){
            for (int j=i+1; j<nums.length; j++){
                if (nums[i] + nums[j] == target){
                    return new int[]{i, j};
                } //end-if
            } //end-for
        } //end-for

        return new int[]{-1, -1};  // No pair adds up to the target
    } //end-twoSum
} // end-solution

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/20-Set-Map/LeetCode/LeetCode-0001-1.cpp
class Solution {
public:
    /// O(n^2) algorithm with O(1) extra space: Check all (i, j) pairs
	vector<int> twoSum(vector<int>& nums, int target) {
        for (int i=0; i<nums.size()-1; i++){
            for (int j=i+1; j<nums.size(); j++){
                if (nums[i] + nums[j] == target){
                    return vector<int>{i, j};
                } //end-if
            } //end-for
        } //end-for

        return {-1, -1};  // No pair adds up to the target
    } //end-twoSum
};
